### PR TITLE
add visibility distance checking to dyno resolver

### PR DESCRIPTION
**Pull request by arezaii**
This is part 2 of a series that implements some disambiguation rules in Dyno that are already updated in the production compiler (via https://github.com/chapel-lang/chapel/pull/20528).

This PR builds on https://github.com/chapel-lang/chapel/pull/23668, specifically this portion of the changes related to relative visibility distance of a call candidate:
>If any candidate is more visible (or shadows) another candidate, discard all candidates that are less visible than (or shadowed by) another candidate.

[reviewed by @mppf - thanks!]

### CODE REVIEW

File: frontend/lib/resolution/disambiguation.cpp [hunk 1]
```diff
@@ -708,6 +738,198 @@ checkVisibilityInVec(Context* context,
...
 }
 
+//
+// helper routines for isMoreVisible (below);
+//
+static bool
+isDefinedInBlock(Context* context, const Scope* scope,
+                 const TypedFnSignature* fn) {
+  LookupConfig onlyDecls = LOOKUP_DECLS | LOOKUP_METHODS;
+  //TODO: should this also/only use LOOKUP_INNERMOST?
```
Comment (mppf) [inline frontend/lib/resolution/disambiguation.cpp:748]: no, because then `id == fn->id()` might not match below, because we might not find that match

**Review by mppf — APPROVED** [review]
Thanks for working on it. It would be nice to add some testing of the new disambiguation rules, but I think it'd be reasonable to put that off until we have more of the disambiguation changes ported over. (And of course, at some point, we will be able to run normal testing with the dyno resolver etc).

File: frontend/lib/resolution/disambiguation.cpp (hunk 1 above)
Comment (arezaii) [reply frontend/lib/resolution/disambiguation.cpp:748]: will remove in a later, trivial PR

Comment (arezaii) [reply frontend/lib/resolution/disambiguation.cpp:748]: fixed in https://github.com/chapel-lang/chapel/pull/23838

**Pull request merged**